DiscordBot.Commands/HelpCommand.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Discord;
+using Discord.WebSocket;
+using DiscordBot.Commands.Core;
+using DiscordBot.Commands.Core.CommandAttributes;
+using DiscordBot.Commands.Core.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DiscordBot.Commands
+{
+    [Name("help")]
+    [Description("Prints help command")]
+    public class HelpCommand : Command 
+    {        
+        private readonly ILogger<HelpCommand> _logger;
+
+        public HelpCommand(ILogger<HelpCommand> logger)
```

Comment:
@Niewidzialny84 What if we add `Lazy<IEnumerable<ICommand>>` to constructor and register this command separately using container's resolve function. The registration would look something like this
```
builder.RegisterType<HelpCommand>().As<ICommand>()
    .WithParameter(
    (pi, ctx) => pi.ParameterType == typeof(Lazy<IEnumerable<ICommand>>),
    (pi, ctx) => new Lazy<IEnumerable<ICommand>>(() => ctx.Resolve<IEnumerable<ICommand>>())
    );
```
Only remember to exclude this command from current registration (you can use `Except<HelpCommand>()`)

Reply:
Changes were applied and solution works correctly.